### DISCUSSION TITLE
Intent Classification

### DISCUSSION THREAD

**Discussion by concurrentprocessingexception** [spring-projects/spring-ai#544] [Q&A]
I am working on the chatbot application for a logistics company.

Scenario:

The idea for the chatbot is get shipment information ( estimated delivery date etc. ) which are available as a custom API, which I have hooked as a function call along with a OpenApiClient and is working.
Given a query from customer "What is the estimated delivery date for my shipment with id XYZ", will make a call to API and respond to user as "The estimated delivery date is 01-01-2025".
The second part is for chatbot to answer general queries about "What is the cancellation policy for my shipment?". These scenarios are covered by RAG model. I have created vectors form the documentation. and is working as well.

Now the problem statement is:

For a given query "I want to information about my shipment", will makes a call to API as well as do a vector search. This looks to me as scenario with "Insconsistent Intent". How do we tell the model to not make API call OR vector database search based on the query? Say if a user is interested in asking for cancellation policy of the shipment then the OpenApiClient should not make a call to API and vice versa.

Any thoughts on How to achieve this with Spring AI.

Food For Thought: Langchain does similar things by defining the router chains.1

**Comment by izeye** [comment]
This seems to be a duplicate of https://github.com/spring-projects/spring-ai/discussions/543.
